File: timus/1005.py
#!/usr/bin/python3

n = int(input())
stones_weight = [int(i) for i in input().split()]
#stones_weight.sort()
#print(stones_weight)

def find_sum_of_combination_weights(n, stones_weight):
    weights = [0 for i in range(2**n)]
#    print(len(weights))
    for i in range(n):
        j = 0
        last_sum_calculated_at = 2 ** i - 1
        #next_sum_position_gap = 2 ** i
        next_sum_position_gap = last_sum_calculated_at + 1
        while j <= last_sum_calculated_at:
            weights[j + next_sum_position_gap] = weights[j] + stones_weight[i]
            j += 1
    return weights




a = find_sum_of_combination_weights(n, stones_weight)
#a = list(set(a))
a.sort()
#print(a)
#print(type(a))
#print(a)

def find_min_weight_difference_of_2_piles(stones_weight, all_possible_stone_piles_weight_sum):
    total_weight = sum(stones_weight)
    minimum_weight_difference = total_weight
    half_weight_sum = total_weight // 2 + 1
    #print(half_weight_sum)
    i = 0
    while i < len(all_possible_stone_piles_weight_sum) and all_possible_stone_piles_weight_sum[i] <= half_weight_sum:
        d = abs(total_weight - 2 * all_possible_stone_piles_weight_sum[i])
        if d < minimum_weight_difference:
            minimum_weight_difference = d
        i += 1
    return minimum_weight_difference

result = find_min_weight_difference_of_2_piles(stones_weight, a)
print(result)

File: contest_sites/timus/vol1/1005.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

vector<int> find_sum_of_combination_of_weights(int number_of_stones, vector<int> stones_weight)
{
  /*
  cout << "In find_sum_of_combination_of_weights function" << endl;
  cout << number_of_stones << endl;
  for (vector<int>::iterator it = stones_weight.begin(); it != stones_weight.end(); ++it){
    cout << *it << ' ';
  }
  cout << endl;

  cout << "Testing: " << (1<<0) << endl;
  */

  vector<int> weights;
  weights.push_back(0);
  for (int i = 0; i < number_of_stones; ++i){
    int next_sum_position_gap = 1 << i;
    for (int j = 0; j < next_sum_position_gap; ++j){
      weights.push_back(weights[j] + stones_weight[i]);
    }
  }

  return weights;
}


int find_min_weight_difference_of_2_piles(vector<int> stones_weight, vector<int> all_possible_stone_piles_weight_sum)
{
  int total_weight = 0;
  for (unsigned int i = 0; i < stones_weight.size(); ++i)
    total_weight += stones_weight[i];

  int minimum_weight_difference = total_weight;
  int diff;

  for (unsigned int i = 0; i < all_possible_stone_piles_weight_sum.size(); ++i){
    diff = total_weight - 2 * all_possible_stone_piles_weight_sum[i];
    diff = abs(diff);
    if (diff < minimum_weight_difference)
      minimum_weight_difference = diff;
  }

  return minimum_weight_difference;
}


int main()
{
  // Testing start
  // int v = 20 * 100000;
  // cout << v << endl;
  // Testing end

  int number;
  cin >> number;
  //cout << number << endl;

  int stone_weight;

  vector<int> stones;
  for (int i = 0; i < number; ++i){
    cin >> stone_weight;
    stones.push_back(stone_weight);
  }

  /*
  for (int i = 0; i < number; ++i)
    cout << stones[i] << ' ';
  cout << endl;
  */

  vector<int> weights = find_sum_of_combination_of_weights(number, stones);
  /*
  for (vector<int>::iterator it = weights.begin(); it != weights.end(); ++it){
    cout << *it << ' ';
  }
  cout << endl;
  cout << "Minimum Difference: ";
  */

  int diff = find_min_weight_difference_of_2_piles(stones, weights);
  cout << diff << endl;
  return 0;
}
